# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dovecot/dovecot-2.2-0004-9a5b132b6832.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---


# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1366051235 -10800
# Node ID 9a5b132b6832137c5ebedb34001354267b41208e
# Parent  b38d5ebacf25379869b4d5119719f773a26fa077
zlib: Don't crash when trying to use zlib_save for handler that isn't compiled in.

diff -r b38d5ebacf25 -r 9a5b132b6832 src/plugins/zlib/zlib-plugin.c
--- a/src/plugins/zlib/zlib-plugin.c	Mon Apr 15 17:22:06 2013 +0300
+++ b/src/plugins/zlib/zlib-plugin.c	Mon Apr 15 21:40:35 2013 +0300
@@ -295,6 +295,10 @@
 		zuser->save_handler = compression_lookup_handler(name);
 		if (zuser->save_handler == NULL)
 			i_error("zlib_save: Unknown handler: %s", name);
+		else if (zuser->save_handler->create_ostream == NULL) {
+			i_error("zlib_save: Support not compiled in for handler: %s", name);
+			zuser->save_handler = NULL;
+		}
 	}
 	name = mail_user_plugin_getenv(user, "zlib_save_level");
 	if (name != NULL) {

